addons/stock/models/stock_picking.py

```diff
@@ -1092,8 +1092,7 @@ def button_validate(self):
         pickings_not_to_backorder.with_context(cancel_backorder=True)._action_done()
         pickings_to_backorder.with_context(cancel_backorder=False)._action_done()
 
-        if self.user_has_groups('stock.group_reception_report') \
-                and self.picking_type_id.auto_show_reception_report:
+        if self._should_show_report():
```

Comment:
I would rather filter `self` on those having a picking_type_id.auto_show_reception_report. Then get the `lines` from this subset of `self`. This will shorten the diff and ensure the issue won't come again in the future

Reply:
Thanks for your review. I've re worked a bit on the code, and i think that it is actually not a good idea to allow the uses of 'button_validate' on a recordset. Even if we filter the self as you suggested, if all the deliveries fulfill the conditions, we'll still have a recordset, and it will fails on the search.
wh_location_ids = self.env['stock.location']._search([('id', 'child_of', self.picking_type_id.warehouse_id.view_location_id.ids), ('usage', '!=', 'supplier')])
I think it's better to change the code in the industry_fsm_stock to validate each picking one by one, instead of calling the method 'button_validate' on the recordset as we are currently doing. Also, to avoid future error, maybe we should also add an 'ensure_one' at the start of the button_validate function ?
What do you think ?:thinking:

Reply:
The button_validate is intended to works on a record set, actually. It was not some version ago. That's why we maybe forgot some part not well-designed for multi records. But you can validate multiple pickings at once in the list view

Reply:
I think i will apply your first suggestion then. that will solve the traceback for industry_fsm without too much change, and i'll let your team handle the corner cases is that ok with you ?

Reply:
Perfect :) thank you